crates/conjure_core/src/ast/expressions.rs

```diff
@@ -370,8 +385,9 @@ impl Expression {
             Expression::FlatProductEq(_, _, _, _) => Some(Domain::BoolDomain),
             Expression::FlatWeightedSumLeq(_, _, _, _) => Some(Domain::BoolDomain),
             Expression::FlatWeightedSumGeq(_, _, _, _) => Some(Domain::BoolDomain),
-            // #[allow(unreachable_patterns)]
-            // _ => bug!("Cannot calculate domain of {:?}", self),
+            Expression::Abs(_, a) => a
+                .domain_of(vars)?
+                .apply_i32(|a, _| Some(a.abs()), &a.domain_of(vars)?),
```

Comment:
so does this apply abs to all ints in the domain? what would happen to something like int(-7..-2)? it should be int(2..7).

also int(-2..3) could be turned into int(2..3) if we are not careful, though it should be int(0..3)

alternatively we can convert int(-2..3) to int(-2,-1,0,1,2,3), then map abs, sort, deduplicate, then we would end up with  int(0,1,2,3). we can optionally convert this back to int(0..3)

this would ve very inefficient with huge domains though, so not ideal.

Reply:
> so does this apply abs to all ints in the domain? what would happen to something like int(-7..-2)? it should be int(2..7).

IIrc, this function deals with the bounds flipping internally.
We already use it for numerous other operations where the bounds can flip.

Reply:
How about the second part? Can you add a test case with -2..3 in it to see what happens please?